src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -153,11 +208,24 @@ export class SavedObjectsClient {
       type: this._type,
       index: this._index,
       refresh: 'wait_for',
+      ignore: [404],
     });
 
-    if (response.found === false) {
-      throw errors.decorateNotFoundError(Boom.notFound());
+    const deleted = response.result === 'deleted';
+    if (deleted) {
+      return {};
     }
+
+    const docNotFound = response.result === 'not_found';
+    const indexNotFound = response.error && response.error.type === 'index_not_found_exception';
+    if (docNotFound || indexNotFound) {
+      // see "404s from missing index" above
+      throw errors.createGenericNotFoundError();
```

Comment:
I also fixed the `lib/errors` module. Including a `decorateNotFoundError` contradicts the goal of not exposing where NotFound errors come from, so now it has a `createGenericNotFoundError()` that does the same thing these lines were doing.

Reply:
:+1: